packages/broker/src/plugins/storage/Storage.ts

```diff
@@ -277,35 +278,51 @@ export class Storage extends EventEmitter {
         return resultStream
     }
 
-    _fetchFromMessageRefForPublisher(streamId: string, partition: number, fromTimestamp: number, fromSequenceNo: number|null, publisherId?: string|null) {
-        const resultStream = this._createResultStream()
-
-        const query1 = 'SELECT payload FROM stream_data WHERE stream_id = ? AND partition = ? AND bucket_id IN ? AND ts = ? AND sequence_no >= ? AND publisher_id = ? ALLOW FILTERING'
-        const query2 = 'SELECT payload FROM stream_data WHERE stream_id = ? AND partition = ? AND bucket_id IN ? AND ts > ? AND publisher_id = ? ALLOW FILTERING'
+    private fetchFromMessageRefForPublisher(
+        streamId: string,
+        partition: number,
+        fromTimestamp: number,
+        fromSequenceNo: number | null,
+        publisherId?: string | null
+    ) {
+        const resultStream = this.createResultStream()
+
+        const query1 = [
+            'SELECT payload FROM stream_data',
+            'WHERE stream_id = ? AND partition = ? AND bucket_id IN ?',
+            'AND ts = ? AND sequence_no >= ? AND publisher_id = ?',
+            'ALLOW FILTERING'
+        ].join(' ')
+        const query2 = [
+            'SELECT payload FROM stream_data',
+            'WHERE stream_id = ? AND partition = ? AND bucket_id IN ? AND ts > ? AND publisher_id = ?',
+            'ALLOW FILTERING'
+        ].join(' ')
 
         this.bucketManager.getBucketsByTimestamp(streamId, partition, fromTimestamp).then((buckets: Bucket[]) => {
             if (buckets.length === 0) {
-                resultStream.push(null)
+                resultStream.end()
                 return
             }
 
             const bucketsForQuery = bucketsToIds(buckets)
 
             const queryParams1 = [streamId, partition, bucketsForQuery, fromTimestamp, fromSequenceNo, publisherId]
             const queryParams2 = [streamId, partition, bucketsForQuery, fromTimestamp, publisherId]
-            const stream1 = this._queryWithStreamingResults(query1, queryParams1)
-            const stream2 = this._queryWithStreamingResults(query2, queryParams2)
+            const stream1 = this.queryWithStreamingResults(query1, queryParams1)
+            const stream2 = this.queryWithStreamingResults(query2, queryParams2)
 
             return pipeline(
-                // @ts-expect-error TODO is cassandraStream EventEmitter or ReadableStream?
                 merge2(stream1, stream2, {
-                    pipeError: true
+                    // @ts-expect-error options not in type
+                    highWaterMark: 16,
+                    pipeError: true,
                 }),
                 resultStream,
                 (err: Todo) => {
-                    if (err) {
-                        resultStream.destroy(err)
-                    }
+                    resultStream.destroy(err || undefined)
+                    stream1.destroy(err || undefined)
```

Comment:
I'm not 100% sure these other destroy calls are necessary, have left them in as it's probably better to explicitly clean up than to not?